karapace/schema_models.py

```diff
@@ -71,6 +77,10 @@ class SchemaType(str, Enum):
     PROTOBUF = "PROTOBUF"
 
 
+def _assert_never(schema_type: SchemaType) -> None:
```

Comment:
```suggestion
def _assert_never(schema_type: NoReturn) -> None:
```

This needs to be `{typing,typing_extensions}.{Never,NoReturn}` to have the desired effect with mypy.

Reply:
I believe `typing.NoReturn` exists for Python 3.7 too, but if it doesn't we could hack it with combination of:

- `from __future__ import annotations`
- `if TYPE_CHECKING: from typing import NoReturn`

Reply:
Done.

Reply:
This looks like the parameter type is still `SchemaType` instead of `NoReturn`. A note that I forgot to mention yesterday, is that mypy doesn't fully understand `in` checks for narrowing, so at the call site, this probably gives a type error with `_assert_never(schema_type: NoReturn) -> ...`:

```python
if schema_type in [SchemaType.AVRO, SchemaType.JSONSCHEMA]:
```

It needs to be changed into individual checks to work:

```python
if schema_type is SchemaType.AVRO or schema_type is SchemaType.JSONSCHEMA:
```

I'm fine if you just want to roll back here too.